components/script/dom/htmlimageelement.rs

```diff
@@ -780,6 +794,61 @@ impl LayoutHTMLImageElementHelpers for LayoutJS<HTMLImageElement> {
     }
 }
 
+//https://html.spec.whatwg.org/multipage/#parse-a-sizes-attribute
+pub fn parse_a_sizes_attribute(input: DOMString, width: Option<u32>) -> Vec<Size> {
+    let mut sizes = Vec::<Size>::new();
+    let unparsed_sizes = input.split(',').collect::<Vec<_>>();
+    for unparsed_size in &unparsed_sizes {
```

Comment:
Since `unparsed_sizes` is used only in the loop, let's iterate on `input.split` directly (which avoids an iteration and vector allocation).

Reply:
Not sure what you want me to do here. Would you please elaborate little more. Thanks :)

Reply:
I meant,

``` rust
for unparsed_size in input.split(',') {
```

This way, we perform lazy iteration.